### ISSUE TITLE
Trying "conda skeleton cran data.tree" gives "Error: Could not parse metadata (Enhances:)"

### ISSUE THREAD

**Issue by kalefranz** [conda/conda-build#1787]
_From @JustGitting on March 3, 2017 2:2_

Hi,

I'm trying to install the R package [data.tree](https://cran.r-project.org/web/packages/data.tree/) using the method described [here](https://www.continuum.io/blog/developer-blog/conda-data-science).

I've successfully used it to install a number of R packages not in the conda repository. However, I'm unable to install the data.tree package.

```
C:\Users\user1>conda skeleton cran "data.tree"
Fetching metadata from https://cran.r-project.org/
Error: Could not parse metadata (Enhances:)
```

It would appear to be a parsing problem, because replacing the "." in "data.tree" produces a different error.

```
C:\Users\user1>conda skeleton cran "data-tree"
Fetching metadata from https://cran.r-project.org/
Package data-tree not found
```
or

```
C:\Users\pc83>conda skeleton cran data..tree
Fetching metadata from https://cran.r-project.org/
Package data..tree not found

C:\Users\pc83>conda skeleton cran data\.tree
Fetching metadata from https://cran.r-project.org/
Package data\.tree not found

C:\Users\user1>conda skeleton cran "data/.tree"
Fetching metadata from https://cran.r-project.org/
Package data/.tree not found
```

System info:
```
C:\>conda info
Current conda install:
               platform : win-64
          conda version : 4.3.13
       conda is private : False
      conda-env version : 4.3.13
    conda-build version : 2.1.5
         python version : 3.6.0.final.0
       requests version : 2.13.0
       root environment : C:\Data\AppData\Local\Continuum\Anaconda3  (writable)
    default environment : C:\Data\AppData\Local\Continuum\Anaconda3
       envs directories : C:\Data\AppData\Local\Continuum\Anaconda3\envs
                          C:\Users\user1\AppData\Local\conda\conda\envs
                          C:\Users\user1\.conda\envs
          package cache : C:\Data\AppData\Local\Continuum\Anaconda3\pkgs
                          C:\Users\user1\AppData\Local\conda\conda\pkgs
           channel URLs : https://conda.anaconda.org/bioconda/win-64
                          https://conda.anaconda.org/bioconda/noarch
                          https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/win-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
                          https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/r/win-64
                          https://conda.anaconda.org/r/noarch
                          https://conda.anaconda.org/anaconda-fusion/win-64
                          https://conda.anaconda.org/anaconda-fusion/noarch
            config file : C:\Users\user1\.condarc
           offline mode : False
             user-agent : conda/4.3.13 requests/2.13.0 CPython/3.6.0 Windows/7 Windows/6.1.7601
```

Not sure if this is an error on my part or a bug in conda.

Thanks.

_Copied from original issue: conda/conda#4784_

**Comment by msarahan** [comment]
@justgitting yes, this is one of those annoying cross-language details.  In Python, a period has very different meaning from R.

That error seems unrelated to the period, though - just a coincidence.  It's more that there seems to be some kind of entry in the metadata that conda-build can't grok.  That error is coming from https://github.com/conda/conda-build/blob/master/conda_build/skeletons/cran.py#L299 - if you look just above, you see this line:

```
(k, v) = line.split(': ', 1)
```

I think that is failing because in your case, there is nothing after the ``:``.  I can't tackle this immediately, but I'd welcome a PR.  I think a solution would be to just skip uninterpretable metadata and log a warning about the skipped metadata, rather than raising that exception.

**Comment by jdblischak** [comment]
@msarahan I think this Issue can be closed now that `conda skeleton cran` can handle blank entries in CRAN metadata fields.

To anyone that finds this Issue because they are receiving this error about parsing the metadata, the first thing to try is update conda-build to the latest version:

```
conda update conda-build
```

**Comment by msarahan** [comment]
Wait - this probably isn't fixed quite yet - I need to tag 3.0.23 with @jdblischak's fix in #2393.  I'll close this since the PR is merged, but 3.0.23 will be the magic number for this to be fixed.

**Comment by jdblischak** [comment]
@msarahan It's safe to close this Issue because this particular problem has actually been fixed since version [3.0.2](https://github.com/conda/conda-build/blob/master/CHANGELOG.txt#L431). My PR was cleaning up the two different solutions that had been merged so that the code was easier to understand and maintain in the future.

**Comment by msarahan** [comment]
Cool, many thanks for closing this issue and for housekeeping!
